mmv1/third_party/terraform/services/container/node_config.go.tmpl

```diff
@@ -2029,10 +2029,13 @@ func flattenHostMaintenancePolicy(c *container.HostMaintenancePolicy) []map[stri
 // node pool updates in `resource_container_cluster`
 func nodePoolNodeConfigUpdate(d *schema.ResourceData, config *transport_tpg.Config, nodePoolInfo *NodePoolInformation, prefix, name string, timeout time.Duration) error {
 
-	// Nodepool write-lock will be acquired when update function is called.
+	// Cluster write-lock will be acquired when createOpF is called, and read-lock will be acquired when waitOpF is
```

Comment:
May be worthwhile adding the "why" here. Like, why are we taking the write lock when `createOfF` is called?